lib/include/iot_mqtt.h

```diff
@@ -251,7 +251,7 @@ void IotMqtt_ReceiveCallback( void * pNetworkConnection,
 IotMqttError_t IotMqtt_Connect( const IotMqttNetworkInfo_t * pNetworkInfo,
                                 const IotMqttConnectInfo_t * pConnectInfo,
                                 uint32_t timeoutMs,
-                                IotMqttConnection_t * pMqttConnection );
+                                IotMqttConnection_t * const pMqttConnection );
```

Comment:
Second const is needed to assure that this API (and others like that) have a semantic that allows t create the instance of the struct _in place_

Reply:
- "const type * name" is where you are not allowed to change the contents that are pointed at, but can change the pointer.
 - "type * const name" is where you are allowed to change the contents pointed at, but not allowed to change the pointer.
 - "const type * const name" is where you're not allowed to change the contents that are pointed at, nor are you allowed to change the pointer.

Since the variable is passed on the stack, you probably don't intend to restrict the function's contents from changing the pointer, although, that's not strictly a bad thing.

Reply:
In this case we want "type * const name" because the caller of IotMqtt_Connect may want to use the same pointer he passed in.